LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -252,7 +252,42 @@ public void AssignInstructorsToClassrooms()
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            if (ClassroomDictionary.Count != 0)
+            {
+                int maxCount = 0;
+                foreach (Classroom classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        maxCount = classroom.ItemList.Count;
+                    }
+                }
+            }
+        }
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            int studentCount = AttributeGenerator.random.Next(0, (classroom.MaximumCapacity - classroom.ItemList.Count) + 1);
+            int count = 0;
+            while (count < studentCount)
+            {
+                foreach (Student student in ActiveStudentList)
```

Comment:
The logic is incorrect: **foreach** allows to add you as many students to the class as you have in the ActiveStudentList.

Reply:
Ok.

Reply:
It is not fixed, if you have 100 active student and decide to register 5 to the class, according to your logic, there are very good chances that you will be registering all 100 students. There is no checking for the upper limit.